### PR TITLE
feat(ChatButton): add in `ChatButton` component

### LINKED ISSUE #15621: Add AI enhancements to React `Button` enhancements
Related #15620 

**Tasks** (Phase 1): 
- [x] Rounded corner prop
- [x] New "selected-disabled" state (works like a visited link)
- [x] Custome `Quick action` button for chat

[Figma file](https://www.figma.com/file/7eMfOAyRclD4LuDFwRAa3p/Carbon-for-AI-Sketches?type=design&node-id=1743%3A80479&mode=design&t=Jo9H2opjALVAM2g7-1)

AI Buttons - Phase 1

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/carbon-design-system/carbon/issues/15621

Adds in the `ChatButton` component, which is just a wrapper around `Button` that restricts the component adhere to AI styles

Screenshot 2024-02-07 at 10 30 53 AM

#### Changelog

**New**

- New `unstable__ChatButton` and `unstable__ChatButtonSkeleton` components
- AVT and VRT tests for `ChatButton`

#### Testing / Reviewing

Ensure the new component matches the spec

### CODE REVIEW

**Comment by aagonzales** [comment]
Visually, this looks good! I just want to confirm we've got the interaction of the new state correct. Now that I see it, I'm wondering if it should actually be more of a read-only instead of disabled.

**Review by aagonzales — APPROVED** [review]
If the chat devs are good with this, then I think its good to go 👍

File: packages/styles/scss/components/chat-button/_chat-button.scss [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+@use '../button';
+@use '../../config' as *;
+@use '../../theme' as *;
+@use '../../type' as *;
+@use '../../utilities/convert';
+
+@mixin chat-button {
+  .#{$prefix}--chat-btn {
+    border-radius: convert.to-rem(24px);
+    // transition: box-shadow 1s;
```
Comment (alisonjoseph) [inline packages/styles/scss/components/chat-button/_chat-button.scss]:
```suggestion
```

**Review by alisonjoseph — APPROVED** [review]
LGTM!

**Review by guidari — APPROVED** [review]
LGTM! 🚀

**Pull request merged**